src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -112,7 +113,7 @@ public IEnumerable<User> Users
                 {
                     RequestContext requestContext = new RequestContext(CorrelationId);
                     requestContext.Logger.Info("Token cache is null or empty");
-                    return new List<User>();
+                    return Enumerable.Empty<User>();
```

Comment:
Miscellaneous: use _Enumerable.Empty<>_ for lists that should not have any members.